aerogear/android-sdk-operator-poc
# Operator poc

A followup of #1 

## Changes

Just moved the implementation file around a little bit and added two handler interfaces so it will enable us to mock those in unit tests.

## Verification steps:

* Deploy the android sdk template (`oc create -f extras/android/android-persistent.json`)
* Scale the android sdk pod down
* Build the operator image (`operator-sdk build docker.io/aerogear/android-sdk-operator:dev`)
* Deploy both olm and the operator itself (instructions in the readme file)
* Check the AndroidSdk custom resource it should have the proper ConfigMapName (android-sdk-config)
* The operator should be installing both the sdk manager and the android packages described in the config map
* Once both install and sync pods are done running, check the android pv by scaling the android sdk pod up again
* Try changing the configmap by adding or removing an android package, the operator should catch the change and run the sync pod again.

ping @dimitraz
